TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs

```diff
@@ -91,6 +92,39 @@ protected virtual void OnDidReceiveNodeRefresh(NodeRefreshParams parameter)
             }
         }
 
+        private void ReceivedExtractUseCopiesNotification(NotificationType notificationType, object parameters)
+        {
+            try
+            {
+                OnDidReceiveExtractUseCopies((ExtractUseCopiesParams)parameters);
+            }
+            catch (Exception e)
+            {
+                RemoteConsole.Error(String.Format("Error while handling notification {0} : {1}", notificationType.Method, e.Message));
+            }
+        }
+
+        protected virtual void OnDidReceiveExtractUseCopies(ExtractUseCopiesParams parameter)
+        {
+            var docContext = GetDocumentContextFromStringUri(parameter.textDocument.uri, false);
+            if (docContext.FileCompiler != null && docContext.FileCompiler.CompilationResultsForProgram != null &&
+                docContext.FileCompiler.CompilationResultsForProgram.CopyTextNamesVariations != null)
+            {
+                var copiesName = docContext.FileCompiler.CompilationResultsForProgram.CopyTextNamesVariations.Select(cp => cp.TextNameWithSuffix).Distinct();
+                var missingCopiesParam = new MissingCopiesParams();
+                missingCopiesParam.textDocument = parameter.textDocument;
+                missingCopiesParam.Copies = new List<string>();
```

Comment:
You can use ToList() missingCopiesParam.Copies = docContext.FileCompiler.CompilationResultsForProgram.CopyTextNamesVariations.Select(cp => cp.TextNameWithSuffix).Distinct().ToList();

Reply:
yes